src/components/customer-resource-show.js

```diff
@@ -70,10 +72,25 @@ export default class CustomerResourceShow extends Component {
     });
   };
 
+  // when there are no fields being edited this is equal to 0; when
+  // any number of fields are being edited this is greater than 0;
+  // this number is stored on the instance so that the state will only
+  // update when this count is equal to or greater than 0
+  editCount = 0;
+
+  handleEditing = (edit) => {
+    this.editCount = this.editCount + (edit ? 1 : -1);
```

Comment:
`this.editCount += (edit ? 1 : -1)`

Reply:
Thanks, this is left over from a refactor away from 

```
this.setState(({ isEditing }) => ({
  isEditing: isEditing + (edit ? 1 : -1)
}))
```

I like your short-circuiting `(edit || -1)` to maybe add `1`, but I think using `?` makes it more clear that `edit` is a boolean and not a number. If, for whatever reason, `edit = 2` then that would throw the count off.

That won't be necessary though, because I like your suggestion of turning the uncontrolled fields into controlled ones. 👍